### PR TITLE
Removed deprecated X86XMMBinaryArithmeticAnalyser

### PR DESCRIPTION
**Pull request by 0dvictor**
TR_X86XMMBinaryArithmeticAnalyser has been replaced by
FloatingPointAndVectorBinaryArithmeticEvaluator; removing.

Signed-off-by: Victor Ding <dvictor@ca.ibm.com>

### CODE REVIEW

File: compiler/x/codegen/FPTreeEvaluator.cpp [hunk 1]
```diff
@@ -562,27 +561,10 @@ TR::Register *OMR::X86::TreeEvaluator::fpBinaryArithmeticEvaluator(TR::Node
                                                               bool              isFloat,
                                                               TR::CodeGenerator *cg)
    {
-   static auto Disable3OpForFP = (bool)feGetEnv("TR_Disable3OpForFP");
-   if (!Disable3OpForFP && cg->useSSEForSinglePrecision() && cg->useSSEForDoublePrecision())
+   if (cg->useSSEForSinglePrecision() && cg->useSSEForDoublePrecision())
       {
       return TR::TreeEvaluator::FloatingPointAndVectorBinaryArithmeticEvaluator(node, cg);
       }
-   // Attempt to use SSE/SSE2 instructions if the CPU supports them, and
```
Comment (vijaysun-omr) [inline compiler/x/codegen/FPTreeEvaluator.cpp]: Could you provide a short summary of why this code was never going to be used ?

Comment (0dvictor) [reply compiler/x/codegen/FPTreeEvaluator.cpp]: `FloatingPointAndVectorBinaryArithmeticEvaluator` always generates same or better code than `XMMBinaryArithmeticAnalyser`. `XMMBinaryArithmeticAnalyser` never leverage 3-operand form instruction or the RegMem form instructions.

**Comment by vijaysun-omr** [comment]
@genie-omr build all

**Comment by vijaysun-omr** [comment]
@0xdaryl it would be good to get your opinion on this before I merge. Thanks.

**Comment by andrewcraik** [comment]
@0xdaryl ping?

**Comment by vijaysun-omr** [comment]
@0xdaryl this is waiting on your review.

**Comment by vijaysun-omr** [comment]
@genie-omr build all

**Comment by Leonardo2718** [comment]
@0dvictor seems like there's a merge conflict

@0xdaryl ping

**Comment by 0dvictor** [comment]
Conflict has been resolved.

**Comment by Leonardo2718** [comment]
@genie-omr build all

**Comment by 0xdaryl** [comment]
@genie-omr build all

**Comment by 0dvictor** [comment]
The change has been rebased, and failures should be fixed. I also updated the copyright headers to 2019.

**Comment by 0xdaryl** [comment]
@genie-omr build xlinux,win

**Comment by 0xdaryl** [comment]
@genie-omr build osx

**Comment by 0xdaryl** [comment]
@vijaysun-omr : this should be good to merge once the AppVeyor build completes.

**Comment by vijaysun-omr** [comment]
Checks have passed and reviews are done. Merging.

**Pull request merged**